azimuth/modules/dataset_analysis/syntax_tagging.py

```diff
@@ -58,29 +58,27 @@ def compute(self, batch: Dataset) -> List[TaggingResponse]:  # type: ignore
             # Remove punctuation for word count and smart tags
             tokens = [token.text for token in doc if not token.is_punct]
 
+            if len(tokens) >= syntax_options.long_utterance_min_word:
+                tag[SmartTag.long] = True
+            if len(tokens) <= syntax_options.short_utterance_max_word:
+                tag[SmartTag.short] = True
+
+            sub_toks = [tok for tok in doc if (tok.dep_ in syntax_options.subj_tags)]
+            obj_toks = [tok for tok in doc if (tok.dep_ in syntax_options.obj_tags)]
+            vrb_toks = [tok for tok in doc if (tok.pos_ in self.verb_tags)]
+            if not sub_toks:
+                tag[SmartTag.no_subj] = True
+            if not obj_toks:
+                tag[SmartTag.no_obj] = True
+            if not vrb_toks:
+                tag[SmartTag.no_verb] = True
+
             # Some issues occur with other languages such as french if using doc.sents directly.
             # Hence, we use an English sentencizer that seems to work better for similar languages.
             doc_sentencizer_en = self.spacy_sentencizer_en(clean_utterance(utterance))
             sentence_count = len(list(doc_sentencizer_en.sents))
```

Comment:
Not new, nor really important, but I thought you'd find it interesting: I just found out that the fastest way to count the number of items in an iterator:
```suggestion
            sentence_count = sum(1 for _ in doc_sentencizer_en.sents))
```
It avoids instantiating and populating a list, just to get its length.

Reply:
Interesting! I do think it's harder to read though, I would keep the `len` given the list here won't be huge.